base/compiler/abstractinterpretation.jl

```diff
@@ -820,10 +819,11 @@ function abstract_call_builtin(interp::AbstractInterpreter, f::Builtin, fargs::U
         end
     end
     rt = builtin_tfunction(interp, f, argtypes[2:end], sv)
-    if f === getfield && isa(fargs, Vector{Any}) && la == 3 && isa(argtypes[3], Const) && isa(argtypes[3].val, Int) && argtypes[2] ⊑ Tuple
+    if f === getfield && isa(fargs, Vector{Any}) && la == 3 &&
+       (a2 = argtypes[2]; a2 ⊑ Tuple) &&
+       (a3 = argtypes[3]; isa(a3, Const)) && (idx = a3.val; isa(idx, Int))
```

Comment:
Reorder these statements back? The check for `Const` is cheaper than the lattice evaluation.